File: src/main/java/nodes/NodesDescriptor.java
package nodes;

import java.util.ArrayList;
import java.util.List;

public class NodesDescriptor {

	private final List<SingleNode> nodes;
	
	public NodesDescriptor() {
		nodes = new ArrayList<SingleNode>();
	}
	
	public void addNode(SingleNode node){
		nodes.add(node);
	}
	
	public List<SingleNode> getNodesList(){
		return nodes;
	}
	
}

File: unity3d-src/ISSAC_Project/Assets/Scripts/Node/NodesDescriptor.cs
using System.Collections.Generic;
using UnityEngine;

[System.Serializable]
public class NodesDescriptor
{
	[SerializeField]
	private List<Node> nodes;
    [SerializeField]
    private string type;

	public NodesDescriptor()
	{
		nodes = new List<Node>();
	}

    public void ClearNodes()
    {
        nodes.Clear();
    }

	public void AddNode(Node node)
	{
        nodes.Add(node);
	}

	public static string CreateJsonString(NodesDescriptor nd)
	{
		return JsonUtility.ToJson(nd);
	}

    public void SetType(string type)
    {
        this.type = type;
    }
}
